spec/EventListener/ShippingExportEventListenerSpec.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace spec\BitBag\SyliusPocztaPolskaShippingExportPlugin\EventListener;
+
+use BitBag\SyliusPocztaPolskaShippingExportPlugin\Api\WebClientInterface;
+use BitBag\SyliusPocztaPolskaShippingExportPlugin\EventListener\ShippingExportEventListener;
+use BitBag\SyliusShippingExportPlugin\Entity\ShippingExportInterface;
+use BitBag\SyliusShippingExportPlugin\Entity\ShippingGatewayInterface;
+use BitBag\SyliusShippingExportPlugin\Repository\ShippingExportRepository;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+
+final class ShippingExportEventListenerSpec extends ObjectBehavior
+{
+    function let(
+        Filesystem $filesystem,
+        ShippingExportRepository $shippingExportRepository,
+        WebClientInterface $webClient,
+        FlashBagInterface $flashBag
+    ): void {
+        $shippingLabelsPath = 'labels';
+        $this->beConstructedWith(
+            $filesystem,
+            $shippingExportRepository,
+            $shippingLabelsPath,
+            $webClient,
+            $flashBag
+        );
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(ShippingExportEventListener::class);
+    }
+
+    function it_exports_shipment
```

Comment:
Two additional scenarios are missing

Reply:
Added scenarios with support for assertions and exceptions